doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+	- Remove portmap
+
+*Return dictionary in format*
+
+```
+{"pfc_bitmask": {
+      "pfc_mask": [VALUE],
+      "pfc_rx_mask": [VALUE],
+      "pfc_tx_mask": [VALUE]
+ },
+ "ptf_test_params": {
+    "port_map_file": [VALUE],
+    "server": [VALUE],
+    "server_ports": [VALUE],
+    "non_server_port": [VALUE],
+    "router_mac": [VALUE],
+    "pfc_to_dscp": [VALUE],
+    "lossless_priorities": [VALUE],
+    "lossy_priorities": [VALUE]
+ }
+}
+```
+
+*Get lossless and lossy priorities*
+
+- Get buffer PG keys
+	```buf_pg_keys = docker exec -i database redis-cli --raw -n 4 KEYS *BUFFER_PG*```
+- Get buffer PG profiles
+	```docker exec -i database redis-cli -n 4 HGET {buf_pg_keys} "profile"```
+- Get lossless priorities based on obtained buffer PG profiles
+- Get lossy priorities based on obtained buffer PG profiles
+
+*Get PFC to DSCP mapping*
+
+* Get DSCP to TC map key
+	```dscp_to_tc_key = docker exec -i database redis-cli --raw -n 4 KEYS *DSCP_TO_TC_MAP*```
+* Get DSCP to TC map hash keys
+	```dscp_to_tc_keys= docker exec -i database redis-cli --raw -n 4 HKEYS "{dscp_to_tc_key}"```
+
+* Get DSCP to TC map
+	```dscp_to_tc = docker exec -i database redis-cli -n 4 HGET {dscp_to_tc_keys} {dscp_to_tc_keys_item}```
+* Get PFC to DSCP map
+	```combine dscp_to_tc with dscp_to_tc_keys```
+
+
+### pfc_storm_template (scope="module")
+Creates dictionary which items will be used to start/stop PFC generator on fanout switch.
+Dictionary values depends on fanout HWSKU (MLNX-OS, Arista or others)
+
+*Return dictionary in format*
+```
+{"file": {
+    "pfc_storm_start": [VALUE],
+    "pfc_storm_stop": [VALUE]
+  },
+ "template_params": {
+      "pfc_gen_file": PFC_GEN_FILE,
+      "pfc_queue_index": PFC_QUEUE_INDEX,
+      "pfc_frames_number": PFC_FRAMES_NUMBER,
+      "pfc_fanout_interface": [VALUE],
+      "ansible_eth0_ipv4_addr": [VALUE],
+      "pfc_asym": True
+  }
+}
+```
+
+For HWSKU equal to "MLNX-OS":
+
+      pfc_storm_start = "pfc_storm_mlnx.j2"
+      pfc_storm_stop = "pfc_storm_stop_mlnx.j2"
+
+For HWSKU equal to "Arista":
+
+      pfc_storm_start = "pfc_storm_arista.j2"
+      pfc_storm_stop = "pfc_storm_stop_arista.j2"
+
+
+### pfc_storm_runner (scope="function")
+
+*Setup steps*
+
+* Start PFC generator on fanout switch
+```
+fanout.exec_template(pfc_storm_template["pfc_storm_start"], pfc_storm_template["template_params"])
+```
+
+*Teardown steps*
+
+* Stop PFC generator on fanout switch
+```
+fanout.exec_template(pfc_storm_template["pfc_storm_stop"], pfc_storm_template["template_params"])
+```
+
+ 
+### enable_pfc_asym (scope="function")
+
+*Setup steps*
+
+* Enable asymmetric PFC on all server interfaces
+
+For every server port(setup["server_ports"]) execute command:
+
+```config interface pfc asymmetric [SERVER_PORT] on```
+
+*Teardown steps*
+
+* Disable asymmetric PFC on all server interfaces
+
+For every server port(setup["server_ports"]) execute command:
+
+```config interface pfc asymmetric [SERVER_PORT] off```
+
+## Test
+
```

Comment:
Please add some brief explanation for the whole test in general

Reply:
Fixed.